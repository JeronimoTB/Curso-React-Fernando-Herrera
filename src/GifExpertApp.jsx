import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Comida']);

    console.log(categories);

    const onAddCategory = (newCategory) => {
        // console.log(newCategory);

        if (categories.includes(newCategory)) return

        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            <h1>GiftExpertApp</h1>

            <AddCategory
                // setCategories={setCategories}
                onNewCategory={event => onAddCategory(event)}
            />

            {/* <button onClick={() => onAddCategory('Naruto')} >Agregar</button> */}

            {
                categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))
            }

        </>
    )
}
