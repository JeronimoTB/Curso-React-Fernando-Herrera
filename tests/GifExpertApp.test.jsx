import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en GifExpertApp', () => { 

    test('debe de hacer match con el snapshot', () => {

        const {container} = render(<GifExpertApp />)
        expect(container).toMatchSnapshot(); 
    })

    test('debe de mostrar una categoria', () => { 

        render(<GifExpertApp />)
        const categoryHeader = screen.getByRole('heading', { level: 3, hidden: true });
        expect(categoryHeader.textContent).toContain('Comida');
     })

})

    