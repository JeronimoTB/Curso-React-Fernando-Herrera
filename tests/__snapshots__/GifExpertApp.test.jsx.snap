// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Comida
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
